libs/common/src/lib/media/media.service.ts

```diff
@@ -26,6 +36,81 @@ export class MediaService {
     });
   }
 
+  async getProjectForModel(model: MediaOwnerModel) {
+    console.log("model", model);
+    let project: Project | null = null;
+    if (model instanceof Project) {
+      project = model;
+    } else if (model instanceof Site || model instanceof Nursery || model instanceof ProjectReport) {
+      project = await Project.findOne({ where: { id: model.projectId } });
+    } else if (model instanceof SiteReport) {
+      const site = await Site.findOne({ where: { id: model.siteId }, attributes: ["projectId"] });
+      project = await Project.findOne({ where: { id: site?.projectId } });
+    } else if (model instanceof NurseryReport) {
+      const nursery = await Nursery.findOne({ where: { id: model.nurseryId }, attributes: ["projectId"] });
+      project = await Project.findOne({ where: { id: nursery?.projectId } });
+    }
+    console.log("project");
+    if (project == null) throw new BadRequestException(`Media is not part of a project.`);
+    return project;
+  }
+
+  async updateCover(media: Media, project: Project) {
+    await Media.update(
```

Comment:
When we chatted about this in Slack, I believe I mentioned that the endpoint should return all updated media in the `included` array of the JSON API response so that the client side cache for these models is updated.